warplib/maths/bytes_access.cairo

```diff
@@ -0,0 +1,83 @@
+from warplib.maths.pow2 import pow2
+from starkware.cairo.common.bitwise import bitwise_and
+from warplib.maths.utils import narrow_safe
+
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.math_cmp import is_le
+
+from starkware.cairo.common.serialize import serialize_word
+
+func byte_accessor{range_check_ptr}(pos : felt) -> (offset : felt):
+    let (le) = is_le(pos, 256)
+    assert le = 1
+
+    let (offset) = pow2(pos)
+    let (offset1) = pow2(pos + 1)
+    let (offset2) = pow2(pos + 2)
+
+    # placing more than 3 bits will exceed the range of felt
+    # and will produce overflowing results
+    if pos == 31 * 8:
+        return (offset + offset1 + offset2)
+    end
+
+    let (offset3) = pow2(pos + 3)
+    let (offset4) = pow2(pos + 4)
+    let (offset5) = pow2(pos + 5)
+    let (offset6) = pow2(pos + 6)
+    let (offset7) = pow2(pos + 7)
+    return (offset + offset1 + offset2 + offset3 + offset4 + offset5 + offset6 + offset7)
+end
+
+func byte_at_index_uint256{bitwise_ptr : BitwiseBuiltin*, range_check_ptr}(
```

Comment:
These functions need to validate their input based on the width of the `bytesX` type at play. You could either add a width argument to check against or extend generateWarplib.ts to generate functions for each width of FixedBytesType